README.md

```diff
@@ -1,81 +1,106 @@
 
-The `install.sh` script is designed to work on Ubuntu (e.g. on the [Travis-CI](https://travis-ci.org/CFD-GO/TCLB) VMs). 
-The `install.sh` script should work on CentOS.
-You can install the **`sudo`** parts by yourself, and use script to install R packages: rdep and rpython.
+### Dependencies
 
+For the code to compile and work you'll need a few things:
+- [R](https://www.r-project.org/) and some R packages ([optparse](https://cran.r-project.org/package=optparse), [numbers](https://cran.r-project.org/package=numbers), [rtemplate](https://github.com/llaniewski/rtemplate), [gvector](https://github.com/llaniewski/gvector), [polyAlgebra](https://github.com/llaniewski/polyAlgebra))
+- [MPI](https://en.wikipedia.org/wiki/Message_Passing_Interface). We recommend [OpenMPI](http://www.open-mpi.org/)
+- If you want to use your GPU, you'll need [nVidia CUDA](https://developer.nvidia.com/cuda-zone)
+- If you want to integrate TCLB with Python, you'll need [python](https://www.python.org/), [numpy](http://www.numpy.org/) with headers and stuff
+- If you want to develop a model using Python, you'll need [python](https://www.python.org/), [sympy](http://www.sympy.org/) and R package: [rPython](https://cran.r-project.org/package=rPython)
 
-### Compilation
-This should work:
+You can install many of these with the provided `tools/install.sh` script (note it requires sudo):
 ```bash
-module load mpi/openmpi-x86_64 # only on CentOS
-make configure
-./configure --enable-double --enable-graphics --with-cuda-arch=sm_20 
-# only CPU ./configure --enable-double --disable-cuda
-make d2q9
+sudo tools/install.sh essentials   # Installs essential system packages needed by TCLB
+sudo tools/install.sh r            # Installs R
+sudo tools/install.sh openmpi      # Installs OpenMPI
+     tools/install.sh rdep         # Installs needed R packages
+sudo tools/install.sh cuda         # Installs CUDA (we recommend to do it on your own)
+sudo tools/install.sh python-dev   # Installs Python libraries with headers
 ```
 
-## Usage
+You can run the `tools/install.sh` script with the `--dry` option, which will print the commands to run, so you can run them on your own.
+*We do not recommend running anything with sudo without checking*
 
-This should also work:
+### `develop` Branch:
+If you want a more recent version, you could try the development branch with `git checkout develop`
+
+### CPU
+To compile the code for CPU, you can use the `--disable-cuda` option for `./configure`:
+```
+./configure --disable-cuda
+```
+### Parallel run
+To run TCLB in parallel (both on multiple CPU and multiple GPU), you can use the standard syntax of MPI parallel run:
 ```bash
-CLB/d2q9/main example/flow/2d/karman.xml
+mpirun -np 8 CLB/d2q9/main example/flow/2d/karman.xml
 ```
 
-## Documentation
+### Running of clusters
+To ease the usage of TCLB on HPC clusters (SLURM/PBS), you can use scripts provided in the [TCLB_cluster](https://github.com/CFD-GO/TCLB_cluster) repository.
```

Comment:
"To assist with using TCLB on HPC clusters (SLURM/PBS) there are scripts provided in the [TCLB_cluster](https://github.com/CFD-GO/TCLB_cluster) repository.